ibroadcast-uploader.py

```diff
@@ -287,35 +277,94 @@ def calcmd5(self, filePath="."):
                 m.update(data)
         return m.hexdigest()
 
-    def upload(self):
-        """
-        Go and perform an upload of any files that haven't yet been uploaded
-        """
+    def check_md5(self):
         self.__load_md5_int()
         self.__load_md5_ext()
-        skipped = 0
-        failed = 0
-        uploaded = 0
 
-        for filename in self.progressbar(self.files, "Processing: ", 60):
-            if self.be_verbose:
-                print(filename)
-            
+        calculated = False
+        current_path=''
+        file_list = self.files[:]
+        if not self.be_silent and not self.be_verbose:
+            file_list = self.progressbar(self.files[:], "Calculating MD5 hashes: ", 60)
+
+        for filename in file_list:
             # Get an md5 of the file contents and compare it to whats up
             # there already
             if (not self.no_cache) and filename in self.md5_int:
                 file_md5 = self.md5_int[filename]
             else:
-                if self.be_verbose:
-                    print('Calculating MD5 for file...')
+                calculated = True
+                if not self.be_silent and self.be_verbose:
+                    if os.path.dirname(filename) != current_path:
+                        current_path = os.path.dirname(filename)
+                        print('\nChecking directory %s...' % current_path)
+                    print('Calculating MD5 for file %s... ' % os.path.basename(filename), end='')
                 file_md5 = self.calcmd5(filename)
                 self.md5_int[filename] = file_md5
 
             if file_md5 in self.md5_ext:
-                if self.be_verbose:
-                    print('Skipping - already uploaded.')
-                skipped += 1
-                continue
+                self.skipped_files.append(filename)
+                if not self.be_silent and self.be_verbose and calculated:
```

Comment:
What if the file checksum wasn't calculated (because we already have it)? In verbose mode we should still display the info that we're skipping the file because it was already uploaded, no matter whether we calculated the checksum or not.

Not sure I understand why this `calculated` flag is needed at all. Please clarify.

Reply:
My bad, I added that check because otherwise simply the text "Skipping, already uploaded" would be shown. For some reason I didn't realize at the time that yes, that information should be printed if we're in verbose mode. I'll rework it.